### ISSUE TITLE
[1.8.3] onLogin handler fired twice

### ISSUE THREAD

**Issue by AndyCC25** [meteor/meteor#10853]
Updating the version from 1.8.1 to 1.8.3,  notice that a part of the code that relies in a callback to `Accounts.onLogin()` method from  it’s being called twice. reference to https://forums.meteor.com/t/onlogin-handler-fired-twice-meteor-1-8-3/51224.

- **Version: Meteor**: 1.8.3
- **Last version without the issue**: Meteor 1.8.1
- **OS**: Ubuntu 14.04
- **Expected Behavior**: `onLogin` callback should run once
- **Actual Behavior**: `onLogin` callback runs twice
- **Reproduction**: run the reproduction project and notice that the `onLogin` callback fire twice, can check by a simple ``console.log()``.  Project:  https://github.com/AndyCC25/meteor-login-bug

**Comment by afrokick** [comment]
not stale

**Comment by brendonlamb** [comment]
Still not state in Meteor 1.10.2 - just upgraded, same issue.

**Comment by AndyCC25** [comment]
Also I notice that still happens the two fire

**Comment by AndyCC25** [comment]
a WA could use a `once()` function from underscore

**Comment by namenotrequired** [comment]
It seems like there's a few separate bugs going on here.

## On initial login

A seemingly old bug in `accounts-ui-unstyled/login_buttons_dropdown.js`: if I click on the login button, I login once. But if I submit the form by hitting enter, I trigger two separate event listeners that both try to log me in, and we end up calling `onLogin` twice.

I made a PR to fix it (and a related bug): https://github.com/meteor/meteor/pull/11173

## On load when already logged in

On refresh, it is called once by `loggedInAndDataReadyCallback`, as normal, and once by `_startupCallback` which was introduced in [this PR that was included in Meteor 1.8.2](https://github.com/meteor/meteor/pull/10543)

I assume this is the bug that didn't happen on 1.8.1. I haven't looked into it further yet.

**Comment by santiagopuentep** [comment]
Still happening on Meteor 1.11.

**Comment by filipenevola** [comment]
This should be fixed on Meteor 1.12

**Comment by jasongrishkoff** [comment]
Was optimistic v1.12 would solve my double login issue, but I guess not.
`Accounts.onLogin` still fires twice for me.
But I'm using a different package: `std:accounts-ui`, which isn't part of the official Meteor packages. Will update if I figure it out.

**Comment by filipenevola** [comment]
@namenotrequired maybe @jasongrishkoff's case is a different one?

**Comment by jamesloper** [comment]
Same behavior here as @jasongriskoff - When I log in with my React form, it fires once. When using a resume token it's firing twice in rapid succession. It seems like a race condition to me. The time between the two logins is median 70 to 80 milliseconds but as high as 400 millis.

Would it be possible to reopen this issue?

**Comment by jamesloper** [comment]
This bug might be simpler to find than we though. It looks like the first time it fires, `Meteor.user()` is `undefined` and the second time, it is the actual user document. `Meteor.userId()` is set both times. So it seems the meteor internals are emitting before and after the default `null` sub is ready.

**Comment by santiagopuentep** [comment]
@jamesloper this is exactly the problem I'm having, it fires twice, but the first time Meteor.user() is not available; the second time it its.

**Comment by StorytellerCZ** [comment]
This issue persists as per #11446, which was closed as a duplicate of this one.

**Comment by brucejo75** [comment]
@jasongrishkoff & @namenotrequired you were right on target.
PR #11785 should address it.

Sorry I just noticed this issue.  To speed things up next time:
- Referencing a PR or other issue does not notify the author
- In a case like this it may have been addressed quicker if you had mentioned the author of the offending PR (e.g. @brucejo75)

Sorry I goofed it & sorry I did not see this earlier!  Simple fix.
